Rocket.Chat.ShareExtension/Base/SEViewController.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  SEViewController.swift
+//  Rocket.Chat.ShareExtension
+//
+//  Created by Matheus Cardoso on 3/1/18.
+//  Copyright © 2018 Rocket.Chat. All rights reserved.
+//
+
+import UIKit
+
+enum SEError: Error {
+    case noServers
+    case canceled
+}
+
+class SEViewController: UIViewController, SEStoreSubscriber {
+
+    lazy var activityIndicator: UIActivityIndicatorView = {
+        let activityIndicator = UIActivityIndicatorView(activityIndicatorStyle: UIActivityIndicatorViewStyle.whiteLarge)
+        activityIndicator.frame = CGRect(x: 0, y: 0, width: 80, height: 80)
+        activityIndicator.layer.cornerRadius = 10
+        activityIndicator.backgroundColor = UIColor.black.withAlphaComponent(0.5)
+        activityIndicator.center = view.center
+        view.addSubview(activityIndicator)
+        return activityIndicator
+    }()
+
+    override var shouldAutorotate: Bool {
+        return false
+    }
+
+    var avoidsKeyboard: Bool = true {
+        didSet {
+            if avoidsKeyboard {
+                startAvoidingKeyboard()
+            } else {
+                stopAvoidingKeyboard()
+            }
+        }
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        store.subscribe(self)
+        startAvoidingKeyboard()
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        store.unsubscribe(self)
+        stopAvoidingKeyboard()
+    }
+
+    func stateUpdated(_ state: SEState) {
```

Comment:
Wouldn't be better to extend `SEStoreSubscriber` with a empty default implementation of `stateUpdated`? Or even make only the child view controllers that actually need to implement `stateUpdated` to conform to `SEStoreSubscriber`? 

By doing that we avoid overriding an empty method.